### PR TITLE
Feature/ti2023 12/email confirmation

### PR DESCRIPTION
**Pull request by Filip-B-32**

### CODE REVIEW

File: EventPlannerAPI/BusinessLayer/Services/UserService.cs [hunk 1]
```diff
@@ -35,7 +38,20 @@ public async Task<IdentityResult> CreateUserAsyncLogic(UserDto newUser)
             var user = _mapper.Map<EventPlannerUser>(newUser);
             try
             {
-                return await _userRepository.CreateUserAsync(user, newUser.Password);
+                var baseUrl = _configuration[SolutionConfigurationConstants.FrontendBaseUrl];
+                var userCreated = await _userRepository.CreateUserAsync(user, newUser.Password);
+
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/Services/UserService.cs:49]: What happens if userCreated is null?

File: EventPlannerAPI/BusinessLayer/Services/UserService.cs [hunk 2]
```diff
@@ -56,6 +72,7 @@ public async Task<string> SendPasswordResetLinkAsync(ForgotPasswordDto forgotPas
                 }
 
                 var token = await _userRepository.GeneratePasswordResetTokenAsync(user);
+
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/Services/UserService.cs:80]: Remove empty space

File: EventPlannerAPI/BusinessLayer/Services/UserService.cs [hunk 3]
```diff
@@ -71,6 +88,28 @@ public async Task<string> SendPasswordResetLinkAsync(ForgotPasswordDto forgotPas
...
+                    return IdentityResult.Failed(error); ;
+                }
+
+                var result = await _userRepository.ConfirmEmailAsync(user, HttpUtility.UrlDecode(confirmEmailDto.Token));
+                return result;
+
+            }
+            catch (Exception ex) 
+            {
+                throw new Exception (ex.Message);
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/Services/UserService.cs]: Why throw a new exception?

File: EventPlannerAPI/DataAccessLayer/Repositories/UserRepository.cs [hunk 4]
```diff
@@ -38,12 +41,13 @@ public async Task<IdentityResult> CreateUserAsync (EventPlannerUser user, string
...
 
                 var result = await _userManager.CreateAsync(user, password);
+
                 if (result.Succeeded)
                 {
                     await _userManager.AddToRoleAsync(user, userRole);
                 }
                 return result;
-
+
```
Comment (HellHound97) [inline EventPlannerAPI/DataAccessLayer/Repositories/UserRepository.cs]: Why empty space?

File: EventPlannerAPI/EventPlannerAPI/Program.cs [hunk 5]
```diff
@@ -45,11 +45,15 @@
...
-	options.User.RequireUniqueEmail = true;
+    options.SignIn.RequireConfirmedAccount = false;
+    options.SignIn.RequireConfirmedPhoneNumber = false;
+    options.User.RequireUniqueEmail = true;
+    options.SignIn.RequireConfirmedEmail = true;
 })
     .AddEntityFrameworkStores<EventPlannerContext>()
     .AddDefaultTokenProviders();
 
+
```
Comment (HellHound97) [inline EventPlannerAPI/EventPlannerAPI/Program.cs]: empty space

File: EventPlannerAPI/BusinessLayer/Services/UserService.cs [hunk 6]
```diff
@@ -35,7 +38,20 @@ public async Task<IdentityResult> CreateUserAsyncLogic(UserDto newUser)
...
+                var baseUrl = _configuration[SolutionConfigurationConstants.FrontendBaseUrl];
+                var userCreated = await _userRepository.CreateUserAsync(user, newUser.Password);
+
+                var token = await _userRepository.GenerateConfirmEmailTokenAsync(user);
+                var confirmLink = baseUrl + "/confirm-account?token=" + HttpUtility.UrlEncode(token) + "&email=" + HttpUtility.UrlEncode(user.Email);
+                var mail = MailRequest.ConfirmAccount(user.Email, user.UserName, confirmLink);
+                if (userCreated != null)
+                {
+                    await _mailService.SendEmailAsync(mail);
+
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/Services/UserService.cs]: empty space

File: EventPlannerAPI/BusinessLayer/DTOs/ConfirmEmailDto.cs [hunk 7]
```diff
@@ -0,0 +1,11 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace BusinessLayer.DTOs
+{
+    public class ConfirmEmailDto
+    {
+
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/DTOs/ConfirmEmailDto.cs]: empty space

File: EventPlannerAPI/BusinessLayer/Interfaces/IUserService.cs [hunk 8]
```diff
@@ -6,6 +6,8 @@ namespace BusinessLayer.Interfaces
     public interface IUserService
     {
 		Task<IdentityResult> CreateUserAsyncLogic(UserDto user);
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/Interfaces/IUserService.cs]: padding

File: EventPlannerAPI/BusinessLayer/Interfaces/IUserService.cs [hunk 9]
```diff
@@ -6,6 +6,8 @@ namespace BusinessLayer.Interfaces
...
+
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/Interfaces/IUserService.cs]: empty space

**Comment by Filip-B-32** [comment]
on the last commit ("Cleaned code") i also added the redux missing files. Before the last commit i accidentally forgot to add them.
it's tested and it works as it should

File: event-planner-frontend/src/features/account-confirmation/store/thunks/accountConfirmationThunk.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { ConfirmEmailDto } from 'features/account-confirmation/api/dtos';
+import UserService from 'services/userService';
+
+const userService = new UserService();
+
+export const accountConfirmationThuk = createAsyncThunk(
```
Comment (gheorghe-sava) [inline event-planner-frontend/src/features/account-confirmation/store/thunks/accountConfirmationThunk.ts]:
```suggestion
export const accountConfirmationThunk = createAsyncThunk(
```

**Review by HellHound97 — APPROVED** [review]

**Review by gheorghe-sava — APPROVED** [review]

**Pull request merged**